operator/controller/src/main/java/io/apicurio/registry/operator/resource/LabelDiscriminator.java

```diff
@@ -0,0 +1,40 @@
+package io.apicurio.registry.operator.resource;
+
+import io.apicurio.registry.operator.OperatorException;
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ResourceDiscriminator;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class LabelDiscriminator<R extends HasMetadata>
```

Comment:
question: we are already using the `labelSelector` in the `ControllerConfiguration`, would you mind explaining why we have also a `resourceDiscriminator`?

In case this is needed, is the `labelSelector` redundant?